applications/serial_lte_modem/src/gpio/slm_at_gpio.c

```diff
@@ -0,0 +1,312 @@
+			if (cur->pin == pin) {
+				slm_gpio_pin = cur;
+			}
+		}
+	}
+
+	/* Add GPIO node if node does not exist */
+	if (slm_gpio_pin == NULL) {
+		slm_gpio_pin = (struct slm_gpio_pin_t *)k_malloc(sizeof(struct slm_gpio_pin_t));
+		if (slm_gpio_pin == NULL) {
+			return -ENOBUFS;
+		}
+		memset(slm_gpio_pin, 0, sizeof(struct slm_gpio_pin_t));
+		sys_slist_append(&slm_gpios, &slm_gpio_pin->node);
+	}
+
+	err = gpio_pin_configure(gpio_dev, pin, gpio_flags);
+	if (err) {
+		LOG_ERR("GPIO_0 config error: %d", err);
+		//TODO: free and remove node
+	}
+
+	slm_gpio_pin->pin = pin;
+	slm_gpio_pin->op = op;
+
+	if (op == SLM_GPIOC_OP_DISABLE) {
+		/* Disable interrupt */
+		err = gpio_pin_interrupt_configure(gpio_dev, pin, GPIO_INT_DISABLE);
+		if (err) {
+			LOG_ERR("Interface pin interrupt config error: %d", err);
+			return err;
+		}
+		/* Remove callback */
+		pin_mask &= ~BIT(pin);
+	}
+
+	return err;
+}
+
+int do_gpio_pin_configure_read(void)
+{
+	int err = 0;
+	struct slm_gpio_pin_t *cur = NULL, *next = NULL;
+
+	sprintf(rsp_buf, "\r\n#XGPIOCFG\r\n");
+	rsp_send(rsp_buf, strlen(rsp_buf));
+
+	if (sys_slist_peek_head(&slm_gpios) != NULL) {
+		SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&slm_gpios, cur,
+					     next, node) {
+			if (cur) {
+				LOG_DBG("%hu,%hu", cur->op, cur->pin);
+				sprintf(rsp_buf, "%hu,%hu\r\n", cur->op, cur->pin);
+				rsp_send(rsp_buf, strlen(rsp_buf));
+			}
+		}
+	}
+
+	return err;
+}
+
+int do_gpio_pin_operate(uint16_t op, gpio_pin_t pin, uint16_t value)
+{
+	int ret = 0;
+	struct slm_gpio_pin_t *cur = NULL, *next = NULL;
+
+	if (sys_slist_peek_head(&slm_gpios) != NULL) {
+		SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&slm_gpios, cur,
+					     next, node) {
+			if (cur) {
+				if (cur->pin != pin) {
+					continue;
+				}
+				if (op == SLM_GPIO_OP_WRITE) {
+					LOG_DBG("Write pin: %d with value: %d", cur->pin, value);
+					ret = gpio_pin_set(gpio_dev, pin, value);
+					if (ret < 0) {
+						LOG_ERR("Cannot write gpio");
+						return ret;
+					}
+				} else if (op == SLM_GPIO_OP_READ) {
+					ret = gpio_pin_get(gpio_dev, pin);
+					if (ret < 0) {
+						LOG_ERR("Cannot read gpio high");
+						return ret;
+					}
+					LOG_DBG("Read value: %d", ret);
+					sprintf(rsp_buf, "\r\n#XGPIO: %d,%d\r\n", pin, ret);
+					rsp_send(rsp_buf, strlen(rsp_buf));
+				} else if (op == SLM_GPIO_OP_TOGGLE) {
+					LOG_DBG("Toggle pin: %d", cur->pin);
+					ret = gpio_pin_toggle(gpio_dev, pin);
+					if (ret < 0) {
+						LOG_ERR("Cannot toggle gpio");
+						return ret;
+					}
+				}
+			}
+		}
+	}
+
+	return 0;
+}
+
+/**@brief handle AT#XGPIOCFG commands
+ *  AT#XGPIOCFG=<op>,<pin>
+ *  AT#XGPIOCFG?
+ *  AT#XGPIOCFG=?
+ */
+int handle_at_gpio_configure(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t pin = 0xff, op = 0xff;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		if (at_params_valid_count_get(&at_param_list) == 0) {
+			return -EINVAL;
+		}
+		err = at_params_short_get(&at_param_list, 1, &op);
```

Comment:
I believe that it should be at_param_unsigned_short_get()

Reply:
Change to at_params_unsigned_short_get.